manager/src/main/java/org/apache/hertzbeat/manager/config/PluginConfig.java

```diff
@@ -0,0 +1,177 @@
+import java.util.List;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanNameGenerator;
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.ImportBeanDefinitionRegistrar;
+import org.springframework.core.type.AnnotationMetadata;
+import org.springframework.stereotype.Component;
+
+
+/**
+ * Plugin configuration
+ */
+@Slf4j
+@Component
+public class PluginConfig implements ImportBeanDefinitionRegistrar, ApplicationContextAware {
+    @Autowired
+    private ApplicationContext applicationContext;
+    @Autowired
+    private DefaultListableBeanFactory defaultListableBeanFactory;
+
+    private static final String CLASS_SUFFIX = ".class";
+
+    private static final String BASE_PLUGIN_PATH = "org/apache/hertzbeat/plugin/Plugin";
+
+    private static final String BASE_PATH = System.getProperty("user.dir") + File.separator + "ext-lib";
+
+    private static final String PACKAGE = "org/apache/hertzbeat/plugin";
+
+    private URLClassLoader getClassLoader() {
+        List<String> jars = getAllJars();
+        if (jars == null) {
+            return null;
+        }
+        List<URL> urlList = new ArrayList<>();
+        for (String jar : jars) {
+            try {
+                URL url = new File(jar).toURI().toURL();
+                urlList.add(url);
+            } catch (MalformedURLException e) {
+                log.error("Error converting jar to URL: {}", jar, e);
+            }
+        }
+        if (urlList.isEmpty()) {
+            return null;
+        }
+        URL[] urls = urlList.toArray(new URL[0]);
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        return new URLClassLoader(urls, contextClassLoader);
+    }
+
+
+    private List<String> getAllJars() {
+        log.debug("BASE_PATH:{}", PluginConfig.BASE_PATH);
+        File file = new File(PluginConfig.BASE_PATH);
+        File[] files = file.listFiles();
+        if (files == null) {
+            return null;
+        }
+        return Arrays.stream(files)
+                .filter(f -> f.getName().endsWith(".jar"))
+                .map(File::getAbsolutePath)
+                .collect(Collectors.toList());
+    }
+
+    private List<String> getAllClassNamesFromJar() throws IOException {
```

Comment:
Is it possible to use `SPI` to simplify `classloader`.

Reply:
Please help me modify it.